course/src/isometry.cc

```diff
@@ -0,0 +1,131 @@
+#include <cmath>
+#include <cstdint>
+#include "isometry.h"
+
+namespace ekumen {
+namespace math {
+
+double& Vector3::operator [] (const int index) const {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
+}
+
+double& Vector3::operator [] (const int index) {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
+}
+
+Vector3 Vector3::operator + (const Vector3& v) const {
+    return Vector3(x_ + v.x_, y_ + v.y_, z_ + v.z_);
+}
+
+Vector3 Vector3::operator - (const Vector3& v) const {
+    return Vector3(x_ - v.x_, y_ - v.y_, z_ - v.z_);
+}
+
+Vector3 Vector3::operator * (const Vector3& v) const {
+    return Vector3(x_ * v.x_, y_ * v.y_, z_ * v.z_);
+}
+
+Vector3 Vector3::operator / (const Vector3&v) const {
+    return Vector3(x_ / v.x_, y_ / v.y_, z_ / v.z_);
+}
+
+Vector3 Vector3::operator = (const Vector3& v) {
+    x_ = v.x();
+    y_ = v.y();
+    z_ = v.z();
+    return Vector3(v); 
+}
+
+Vector3& ekumen::math::Vector3::operator += (const Vector3& v) { 
+    x_ += v.x_;
+    y_ += v.y_;
+    z_ += v.z_;
+    return *this;
+}
+
+Vector3& ekumen::math::Vector3::operator -= (const Vector3& v) {
+    x_ -= v.x_;
+    y_ -= v.y_;
+    z_ -= v.z_;
+    return *this; 
+}
+
+Vector3& ekumen::math::Vector3::operator *= (const Vector3& v) {
+    x_ *= v.x_;
+    y_ *= v.y_;
+    z_ *= v.z_;
+    return *this; 
+}
+
+Vector3& ekumen::math::Vector3::operator /= (const Vector3& v) {
+    x_ /= v.x_;
+    y_ /= v.y_;
+    z_ /= v.z_;
+    return *this; 
+}
+
+bool ekumen::math::Vector3::operator == (const Vector3& v) const {
+    return (x_ == v.x_) && (y_ == v.y_) && (z_ == v.z_);
+}
+
+bool ekumen::math::Vector3::operator != (const Vector3& v) const {
+    // introduce tolerance
+    return (x_ != v.x_) || (y_ != v.y_) || (z_ != v.z_);
+}
+
+bool ekumen::math::Vector3::operator == (const std::initializer_list<double>& list) const {
+    //check list correct size (3)
+    //check for loops that avoids SIGSEGV
+    std::initializer_list<double>::iterator it;
+    int i;
```

Comment:
Not very C++-ish. Better create where you use (inside the for) and better yet, avoid using index variables altogether.